src/gpuarray_buffer_cuda.c

```diff
@@ -42,7 +42,7 @@ void *cuda_make_ctx(CUcontext ctx, int flags) {
   cuda_context *res;
   void *p;
 
-  res = malloc(sizeof(*res));
+  res = calloc(1, sizeof(*res));
```

Comment:
This gets rid of undefined memory usage in the cache for kernels with regards to the bin_id.  Since it might not be written to completely, this makes sure that the unwritten parts are zeroed.